File: typescript/src/advertisement/domain/model/Advertisement.ts
export class Advertisement {

  constructor(
    private readonly _id: string,
    private _description: string,
    private _password: string,
    private _date: Date
  ) {
  }

  public update(description: string, password: string): void {
    this._description = description;
    this._password = password;
    this.updateDate();
  }

  public renew(): void {
    this.updateDate();
  }

  public id(): string {
    return this._id
  }

  public description(): string {
    return this._description
  }

  public password(): string {
    return this._password
  }

  public date(): Date {
    return this._date
  }

  private updateDate(): void {
    this._date = new Date();
  }
}

File: kotlin/src/main/kotlin/advertisement/domain/model/Advertisement.kt
package advertisement.domain.model

import advertisement.domain.model.value_object.Password
import java.time.LocalDateTime

data class Advertisement(val id: String?, var description: String?, var password: Password?, var date: LocalDateTime)
{
    fun update(description: String?)
    {
        this.description = description
        updateDate()
    }

    fun renew()
    {
        updateDate()
    }

    private fun updateDate()
    {
        this.date = LocalDateTime.now()
    }
}
